http-sink/metrics.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"net/http"
+)
+
+type MetricsPublisher struct {
+	payloadTotalCounter prometheus.Counter
```

Comment:
Use `promauto.NewCounterVec` to save all the registration?

Reply:
removed all registration functions.